### PR TITLE
Feat/21365 scaffolder custom action bitbucket run pipeline

### LINKED ISSUE #21365: 🚀 Feature: Trigger an bitbucket pipeline with an action similar to github:dispatch
### 🔖 Feature description

i would greatly appreciate if i could trigger an bitbucket cloud pipeline on my backstage template

### this triggers an pipeline on github that creates an S3 Bucket on AWS
We have this on github
    - id: github-action
      name: Trigger GitHub Action
      action: github:actions:dispatch
      input:
        workflowId: s3-aws.yml
        repoUrl: 'github.com?repo=S3-AWS&owner=owner'
        branchOrTagName: 'main'
        workflowInputs:
          bucket_name: ${{ parameters.bucket_name }}
          region: ${{ parameters.region }}
          action: ${{ parameters.action }}
        
i would like to do the same with bitbucket, but i cant seem to find anything on how to do it

### 🎤 Context

My company only uses bitbucket cloud

Im creating a pipeline that have numerous terraform modules

my goal with backstage is to create an infra-as-a-code platform with the terraform modules templates, in a way that my devs can create any infra with my terraform code

i want to publish the content of the template, and then trigger the terraform pipeline with the specific parameters set by the template

### ✌️ Possible Implementation

I would like to pass the repoUrl, or the branch, or make an commit that triggers the pipeline

maybe something like bitbucketCloud:actions:triggerpipeline

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by mattprinold**
## Hey, I just made a Pull Request!
resolves #21365 
Adds a new builtin action to the scaffolder for running bitbucket cloud pipelines

Prerequisites for manual testing:

- bitbucket cloud account 
  - with a project & repo and and pipeline configured

Scaffolder template example:
```yaml
apiVersion: scaffolder.backstage.io/v1beta3
# https://backstage.io/docs/features/software-catalog/descriptor-format#kind-template
kind: Template
metadata:
  name: bitbucket-cloud-pipeline-runner
  title: bitbucket Cloud Pipeline Runner
  description: Run a bitbucket cloud pipeline
spec:
  type: service
  parameters:
    - title: Fill in some steps
      required:
        - workspace
        - repo_slug
      properties:
        workspace:
          title: Workspace
          type: string
        repo_slug:
          title: Repo name
          type: string
        run_type:
          title: Pipeline job type
          type: string
          enum:
            - pipelineForBranch
            - pipelineForCommitOnBranchOrTag
            - specificPipelineDefinitionForCommit
            - specificPipelineDefinitionForCommitOnBranchOrTag
            - customPipelineWithVariables
            - pullRequestPipeline
      dependencies:
        run_type:
          oneOf:
            - properties:
                run_type:
                  enum:
                    - pipelineForBranch
                body:
                  type: object
                  properties:
                    target:
                      type: object
                      properties:
                        ref_type:
                          type: string
                        type:
                          type: string
                        ref_name:
                          type: string
            - properties:
                run_type:
                  enum:
                    - pipelineForCommitOnBranchOrTag
                body:
                  type: object
                  properties:
                    target:
                      type: object
                      properties:
                        ref_type:
                          type: string
                        type:
                          type: string
                        ref_name:
                          type: string
                        commit:
                          type: object
                          properties:
                            type:
                              type: string
                            hash:
                              type: string
            - properties:
                run_type:
                  enum:
                    - specificPipelineDefinitionForCommit
                body:
                  type: object
                  properties:
                    target:
                      type: object
                      properties:
                        type:
                          type: string
                        commit:
                          type: object
                          properties:
                            type:
                              type: string
                            hash:
                              type: string
                        selector:
                          type: object
                          properties:
                            type:
                              type: string
                            pattern:
                              type: string
            - properties:
                run_type:
                  enum:
                    - specificPipelineDefinitionForCommitOnBranchOrTag
                body:
                  type: object
                  properties:
                    target:
                      type: object
                      properties:
                        ref_type:
                          type: string
                        type:
                          type: string
                        ref_name:
                          type: string
                        commit:
                          type: object
                          properties:
                            type:
                              type: string
                            hash:
                              type: string
                        selector:
                          type: object
                          properties:
                            type:
                              type: string
                            pattern:
                              type: string
            - properties:
                run_type:
                  enum:
                    - customPipelineWithVariables
                body:
                  type: object
                  properties:
                    target:
                      type: object
                      properties:
                        ref_type:
                          type: string
                        type:
                          type: string
                        ref_name:
                          type: string
                        selector:
                          type: object
                          properties:
                            type:
                              type: string
                            pattern:
                              type: string
                    variables:
                      type: array
                      items:
                        type: object
                        properties:
                          key:
                            type: string
                          value:
                            type: string
                          secured:
                            type: boolean
            - properties:
                run_type:
                  enum:
                    - pullRequestPipeline
                body:
                  type: object
                  properties:
                    target:
                      type: object
                      properties:
                        type:
                          type: string
                        source:
                          type: string
                        destination:
                          type: string
                        destination_commit:
                          type: object
                          properties:
                            hash:
                              type: string
                        commit:
                          type: object
                          properties:
                            hash:
                              type: string
                        pullrequest:
                          type: object
                          properties:
                            id:
                              type: string
                        selector:
                          type: object
                          properties:
                            type:
                              type: string
                            pattern:
                              type: string

    - title: Authentication
      properties:
        token:
          title: Token
          type: string
          description: Auth JWT token
  steps:
    - id: bitbucket-pipeline-run
      name: Run Bitbucket pipeline
      action: bitbucket:pipelines:run
      input:
        workspace: ${{ parameters.workspace }}
        repo_slug: ${{ parameters.repo_slug }}
        body: ${{ parameters.body }}
        token: ${{ parameters.token }}
  output:
    text:
      - title: More Information
        content: >
          **Build Number:** `${{
          steps['bitbucket-pipeline-run'].output.buildNumber }}`
    links:
      - title: Repository
        url: ${{ steps['bitbucket-pipeline-run'].output.repoUrl }}
      - title: View pipeline
        url: ${{ steps['bitbucket-pipeline-run'].output.pipelinesUrl }}
```

example bitbucket pipeline:
```yaml
image: atlassian/default-image:3

pipelines:
  pull-requests:
    '**':
      - step:
          name: 'Pull request pipeline step 1'
          script:
            - echo "Hello im running from a PR"
  custom:
    test-pipeline:
    - step:
        name: 'Custom pipeline step 1'
        script:
          - echo "Hello im running from a custom pipeline definition"
  default:
    - step:
        name: 'Default pipeline step 1'
        script:
          - echo "Hello im running from your default pipeline
```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by NeuralEvolution** [comment]
@benjdlambert can we get this PR reopened please.

**Comment by benjdlambert** [comment]
@NeuralEvolution done! Apologies - this slipped through without a review over the holidays!

File: .changeset/green-dolphins-cover.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-scaffolder-backend-module-bitbucket': patch
```
Comment (benjdlambert) [inline .changeset/green-dolphins-cover.md:2]: This should be `minor` I think

**Comment by NeuralEvolution** [comment]
@benjdlambert updated the changeset to minor

File: .changeset/green-dolphins-cover.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-scaffolder-backend-module-bitbucket': minor
```
Comment (benjdlambert) [inline .changeset/green-dolphins-cover.md:2]: Think this one should stay as patch as this isn't breaking and the package is 0.x

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Just one thing about the changeset

File: .changeset/green-dolphins-cover.md (hunk 2 above)
Comment (NeuralEvolution) [reply .changeset/green-dolphins-cover.md]: updated

Comment (benjdlambert) [reply .changeset/green-dolphins-cover.md]: @NeuralEvolution sorry - only the `@backstage/plugin-scaffolder-backend-module-bitbucket` should be patch. The `@backstage/plugin-scaffolder-backend` can be minor :)

Comment (NeuralEvolution) [reply .changeset/green-dolphins-cover.md]: @benjdlambert updated

**Review by benjdlambert — APPROVED** [review]
Nice thanks! :tada:

**Pull request merged**